graph_weather/models/atmorep/model/atmorep.py

```diff
@@ -0,0 +1,282 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class AtmoRep(nn.Module):
+    """
+    AtmoRep model that processes multiple input fields (e.g., weather data, satellite images)
+    with both per-field encoders and a shared temporal encoder. The model combines both
+    field-specific information and temporal dependencies to generate predictions.
+
+    Args:
+        config (object): Configuration object containing model parameters like input fields,
+                         model dimensions, and dropout rate.
+    """
+
+    def __init__(self, config):
```

Comment:
Can you add type hints for what type of config this should have? Or add all the config options as arguments, so it can be instantiated with either `AtmoRep(**config)` or by filling in keyword arguments?